examples/evaluators/golang/simple/main.go

```diff
@@ -72,57 +83,107 @@ func main() {
 
 	start := time.Now()
 
+	//proposedMatchIds  []string,
+	//overloadedPlayers map[string][]int
+	//overloadedMatches map[int][]int
+	//approvedMatches   map[int]bool
 	proposedMatchIds, overloadedPlayers, overloadedMatches, approvedMatches, err := stub(cfg, pool)
 	overloadedPlayerList, overloadedMatchList, approvedMatchList := generateLists(overloadedPlayers, overloadedMatches, approvedMatches)
 
-	fmt.Println("overloadedPlayers")
+	lgr.Info("overloadedPlayers")
 	pretty.PrettyPrint(overloadedPlayers)
-	fmt.Println("overloadePlayerList")
+	lgr.Info("overloadePlayerList")
 	pretty.PrettyPrint(overloadedPlayerList)
-	fmt.Println("overloadedMatchList")
+	lgr.Info("overloadedMatchList")
 	pretty.PrettyPrint(overloadedMatchList)
-	fmt.Println("approvedMatchList")
+	lgr.Info("approvedMatchList")
 	pretty.PrettyPrint(approvedMatchList)
 
+	// Choose matches to approve from among the overloaded ones.
 	approved, rejected, err := chooseMatches(overloadedMatchList)
 	approvedMatchList = append(approvedMatchList, approved...)
+	approvedPlayers := make([]string, 0)
+	potentiallyRejectedPlayers := make([]string, 0)
 
 	// run redis commands to approve matches
 	for _, proposalIndex := range approvedMatchList {
 		// The match object was already written by the MMF, just change the
 		// name to what the Backend API (apisrv.go) is looking for.
 		proposedID := proposedMatchIds[proposalIndex]
 		// Incoming proposal keys look like this:
-		// proposal.1542600048.80e43fa085844eebbf53fc736150ef96.testprofile
+		// proposal.80e43fa085844eebbf53fc736150ef96.testprofile
 		// format:
-		// "proposal".timestamp.unique_matchobject_id.profile_name
+		// "proposal".unique_matchobject_id.profile_name
 		values := strings.Split(proposedID, ".")
```

Comment:
Add a precondition to guarantee that the name format is accurate before reading fields based on index. Otherwise the error here is going to be a cryptic error.

Reply:
added